R/zchunk_L100.Population_downscale_ctry.R

```diff
@@ -34,19 +34,129 @@ module_socioeconomics_L100.Population_downscale_ctry_DISABLED <- function(comman
+      select(-pop)
+    # Need to create matching iso codes for three countries in UN, but not Maddison.
+    # Set Serbia and Montenegro ratio equal to Serbia & Montenegro and use Indonesia population ratio for East Timor
+    mne_srb_tls <- filter(maddison_hist_ratio, iso == "idn" | iso == "scg") %>%
+      spread(iso, ratio_iso) %>%
+      rename(mne = scg, tls = idn) %>%
+      mutate(srb = mne) %>%
+      gather(iso, ratio_iso, -year)
+    # Combine with other ratio_iso values
+    maddison_hist_ratio <- bind_rows(maddison_hist_ratio, mne_srb_tls)
+
+    # Fifth, since UN population data are available from 1950+, use Maddison ratios to extrapolate pre-1950 values relative to UN 1950
+    un_pop_clean <- UN_popTot %>%  # Clean raw UN population data
+      filter(Scenario == "EST") %>%  # Historic only (not projections)
+      select(-Region, -Sex, -Scenario) %>%
+      rename(iso = Country, year = Year, pop = Value) %>%
+      mutate(iso = tolower(iso)) %>%
+      filter(year %in% socioeconomics.UN_HISTORICAL_YEARS) %>%  # Keep only the years that we are using
+      mutate(iso = gsub("xea", "twn", iso))  # Correct iso code for Taiwan
+
+    # Sixth, combine with Maddison historic ratios to get values for pre-1950 years
+    L100.Pop_thous_ctry_Yh <- un_pop_clean %>%
+      full_join(maddison_hist_ratio, by = c("iso", "year")) %>%
+      complete(iso, nesting(year)) %>%  # Completes tibble to include all years for all iso
+      group_by(iso) %>%
+      mutate(pop = if_else((year %in% socioeconomics.MADDISON_HISTORICAL_YEARS), (pop[year == socioeconomics.AGG_BASE_YEAR] * ratio_iso), pop)) %>%  # Multiply pop in 1950 by ratio to get prior years' pop
+      filter(iso != "scg") %>%  # "Serbia & Montenegro" separated into individual countries, do not need combined country
+      mutate(pop = if_else(is.na(pop), 0, pop)) %>%  # Per old data system: At this point the remaining mismatched countries are small, so set to zero
+      mutate(year = as.integer(year)) %>%  # To avoid problems later on, save years as integers
+      rename(value = pop) %>%
+      select(-ratio_iso)
+
+    ## (2) SSP population projections by country
+
+    # Final historical population from UN
+    pop_final_hist <- filter(L100.Pop_thous_ctry_Yh, year == socioeconomics.FINAL_HIST_YEAR) %>%
+      rename(pop_final_hist = value) %>%
+      select(-year)
+
+    # First, generate ratios of future population to base year (2010) for all SSPs. The ratios will be applied to the historical year populations so there are no jumps/inconsistencies.
+    L100.Pop_thous_SSP_ctry_Yfut <- as_tibble(SSP_database_v9) %>% # Note units in SSP database are millions, but convert to thousands when we multiply by historic year
+      filter(MODEL == "IIASA-WiC POP" & VARIABLE == "Population") %>%  # IIASA-WiC is the official SSP population data set
```

Comment:
Disagree.  This isn't really a constant; it's extracting a single table from a data structure that contains a whole bunch of tables smooshed together.  Unless there's a reason why we might want to make a coordinated change across several chunks in which table is selected here, there is no reason to clutter `constants.R` with this.